designer/client/inline-conditions.js

```diff
@@ -0,0 +1,200 @@
+import React from 'react'
+import { ConditionsModel, Condition, Field, Value } from './inline-condition-model'
+import { clone } from './helpers'
+
+const conditionalOperators = {
+  'default': {
+    is: {},
+    'is not': {}
+  },
+  'TextField': {
+    matches: {}
+  },
+
+  getConditionals: (fieldType) => {
+    return Object.assign(conditionalOperators[fieldType] || {}, conditionalOperators['default'])
+  }
+}
+
+class InlineConditions extends React.Component {
+  constructor (props) {
+    super(props)
+
+    const { data } = this.props
+
+    this.fields = [].concat.apply([],
+      data.pages.map(page => page.components.filter(component => component.name)
+        .map(component => ({ label: component.title, name: component.name, type: component.type, values: component.options.list ? data.lists.find(it => it.name === component.options.list).items : [] })))
+    ).reduce((obj, item) => {
+      obj[item.name] = item
+      return obj
+    }, {})
+
+    this.state = {
+      conditions: new ConditionsModel()
+    }
+  }
+
+  onClickAddItem = e => {
+    this.setState({
+      condition: {}
+    })
+  }
+
+  onClickAnd = e => {
+    this.setState({
+      condition: {
+        coordinator: 'and'
+      }
+    })
+  }
+
+  onClickOr = e => {
+    this.setState({
+      condition: {
+        coordinator: 'or'
+      }
+    })
+  }
+
+  onClickSave = e => {
+    const { conditions, condition } = this.state
+
+    this.setState({
```

Comment:
i don't think this is doing what you're expecting,  did you meant to do `data.save(..)`? `this.setState` will only change the state of the current component but will be destroyed after it unmounts.

have a look at data.save(..) in other components. it's a method passed down from App (parent component), do the put request to server, and do a setState at the App level so no data is lost on an unmount. (App doesn't unmount)

Reply:
I'm expecting it to save it in the scope of the component, not persist it to the model - I expect that to be done when the user clicks save on the overall containing component. So I *think* it's doing the right thing in this case? 
I'll rename the method as it's definitely a little confusing

Reply:
ah my bad. you'll have to go into `page-create.js` and add to the `onSubmit` call. when `data.save(copy)` is being called at the moment, `copy` is a copy of data, + the new/replaced data to save. At the moment the conditions haven't been added, so it's not being persisted when save is clicked

Reply:
Yep, that's intentional (I haven't gotten round to working out how to persist stuff yet), I'm still working on the UI. I'm just using page-create as a place to play with it and to show you guys what I'm doing.
Out of interest is there a way for me to get ahold of the state of my component in the onSubmit call?

Reply:
What are you trying to achieve? So you can persist the created condition? 
at the moment data.save() is called from onSubmit of the form. so as long as there's an input with `name='condition' value = 'isNewComponent'` You'll be able to retrieve the value with `formData.get('condition')`. 

As for getting the state object, afaik the parent can't access its children's states but you can pass down a method to the child, like `handleConditionChange` that calls `setState`. When the child calls `handleConditionChange` it will change the state of the parent. Ping me on gitter if this wasn't clear! happy to explain.

Reply:
Awesome, that's really helpful thanks!

Reply:
@jenbutongit are you happy for me to resolve this one?